api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/components/Action.java

```diff
@@ -122,4 +122,12 @@ public boolean isRoot() {
     return true;
   }
 
+	private boolean rpcMethodUnsupported(String method) {
+		if (rpc == null || rpc.getHttpMethod() == null || rpc.getHttpMethod().getName() == null) {
+			return false;
```

Comment:
I'm looking at my own PR and thinking maybe this should be returning true here instead of false.

Reply:
It appears so. To avoid the negative logic confusion, I suggest the following:

```
	public OperationResult run(OperationData input) {
		if (firstStep != null && rpcMethodSupported(input.getMethod())) {
			return firstStep.run(input);			
		}
		return new OperationResult(HttpServletResponse.SC_NOT_IMPLEMENTED);

	}
	
	...
	
	private boolean rpcMethodSupported(String method) {
		return rpc != null &&
			rpc.getHttpMethod() != null &&
			rpc.getHttpMethod().getName() != null &&
			rpc.getHttpMethod().getName().equalsIgnoreCase(method);
	}

```

Reply:
Or at least correct such as

```
	public OperationResult run(OperationData input) {
		if (firstStep == null || rpcMethodUnsupported(input.getMethod())) {
			return new OperationResult(HttpServletResponse.SC_NOT_IMPLEMENTED);
		}
		return firstStep.run(input);
	}
	
	...
	
	private boolean rpcMethodUnsupported(String method) {
		return rpc == null ||
			rpc.getHttpMethod() == null ||
			rpc.getHttpMethod().getName() == null ||
			!rpc.getHttpMethod().getName().equalsIgnoreCase(method);
	}
```